### PR TITLE
Compile SDL Mixer X, add HAVE_MIXERX=1 flag to SRB2 compile

### LINKED ISSUE #32: MIDI music not supported
every time I open up the game I'm greeted with
```lua
ERROR: Mix_LoadMUS_RW: Couldn't open /etc/timidity/freepats.cfg
ERROR: Music _title could not be loaded: engine failure!
```
I've installed the necessary dependencies and that file _does_ exist on my system, but the flatpak file for it can't see it. I haven't yet tested this, but I'm pretty sure this is only an issue with the Flatpak bundling...

### PR DESCRIPTION
**Pull request by riomccloud**
Tries to fix #32 
Still testing, not sure it'll work

### CODE REVIEW

File: org.srb2.SRB2.yaml [hunk 1]
```diff
@@ -53,6 +53,33 @@ modules:
...
+    cleanup:
+      - /include
+      - /lib/*.la
+      - /lib/*.so
+      - /lib/pkgconfig
+      - /share/doc
+    sources:
+      - type: git
+        url: https://github.com/WohlSoft/SDL-Mixer-X
+        commit: 7181c771aa29deaea8bfbe6098613c3a929e9189
```
Comment (ghisvail) [inline org.srb2.SRB2.yaml:67]: To track the latest tagged version automatically:

```suggestion
        tag: 2.6.0
        commit: 7181c771aa29deaea8bfbe6098613c3a929e9189
        x-checker-data:
          type: git
          tag-pattern: ^([\d.]+)$
```

Comment (riomccloud) [reply org.srb2.SRB2.yaml:65]: Done

**Comment by riomccloud** [comment]
Not sure what's the problem here... SDL2 and SDL Mixer X seems to be fine, but SRB2 fails to build. IDK if I need to compile [extra components with SDL Mixer X](https://github.com/WohlSoft/SDL-Mixer-X/blob/master/docs/index.md#more-file-formats-support) (and which), or is something related to Flathub's SDL...

**Comment by bitten2up** [comment]
> It compiled normally, but the SRB2 still doesn't play MIDI. For a moment I thought I would need to build additional SDL Mixer X components, but even with them it doesn't work. Now, SRB2 asks for a missing "Timidity" library, that [should be included with base SDL Mixer X](https://github.com/WohlSoft/SDL-Mixer-X/blob/master/docs/index.md#more-file-formats-support).
> 
> image
> 
> During build, tho, it mentions Timidity two times:
> 
> image image
> 
> No idea what's the problem here... @katsy any idea?

you need fluidsynth, give me a sec, im going to try and comment the changes needed

**Review by bitten2up — CHANGES_REQUESTED** [review]
add fluidsynth's sources (hopefully)

**Comment by bitten2up** [comment]
also don't use SDL-Mixer-X on `2.6.0` until 2.2.14, there is a bug that causes it to crash

would of put it on master, but it got put to next instead

**Comment by riomccloud** [comment]
> also don't use SDL-Mixer-X on `2.6.0` until 2.2.14

Sorry, I didn't understand. What do you refer with "2.2.14"?

**Comment by katsy** [comment]
2.2.14 is the next stable release of SRB2. 2.2.13 is the current, and it was discovered that verison 2.6.0 of SDL Mixer X causes a startup crash. It's been fixed upstream for 2.2.14.

So you either need to use an older version of Mixer X or wait an indefinite amount of time for 2.2.14.

**Comment by riomccloud** [comment]
Ah, sure. Thanks.

**Comment by riomccloud** [comment]
Tried building with older SDL Mixer X versions, but no success (2.5.0 causes the same bug as the latest, and 2.2.0 seems to not allow building without additional unnecessary modules. For now, I built the latest version, that will work when SRB2 v2.2.14 releases. Thank you guys for the help! ^^

**Comment by bitten2up** [comment]
0_0

**Comment by riomccloud** [comment]
@ghisvail Oops! Yes, I've closed it by accident.

**Comment by riomccloud** [comment]
@ghisvail Oops! Yes, I've closed it by accident.

File: org.srb2.SRB2.yaml [hunk 2]
```diff
@@ -12,6 +12,7 @@ finish-args:
   - --persist=.srb2
 modules:
   - shared-modules/glu/glu-9.json
+  - shared-modules/SDL2/SDL2-with-libdecor.json
```
[the remark this answers is not in the document]
Comment (ghisvail) [reply org.srb2.SRB2.yaml:15]: You will need to rebase onto the latest commit to benefit from the new runtime. I did a quick test and we now have window decorations (tested on KDE).

**Comment by ghisvail** [comment]
Resumed in #76

**Pull request closed without merging**